File: app/backend/src/main/kotlin/pwr/psi/Powierzenia3000/shared/model/Course.kt
package pwr.psi.Powierzenia3000.shared.model

import javax.persistence.*

@Entity
data class Course(
        @Id @GeneratedValue
        val id: Long = 0,
        val code: String = "",
        val courseType: CourseType = CourseType.WYKLAD,
        @OneToMany(mappedBy = "course")
        val translation: List<CourseTranslation> = listOf()
)


enum class CourseType{
    PROJEKT, LABORATORIA, CWICZENIA, WYKLAD, PRAKTYKI
}

File: app/frontend/src/shared/models/Course.ts
export interface Course {
  id: number;
  code: string;
  courseType: string;
}

export enum CourseType {
  PROJEKT = "Projekt",
  LABORATORIA = "Laboriatoria",
  CWICZENIA = "Ćwiczenia",
  WYKLAD = "Wykład",
  PRAKTYKI = "Praktyki"
}
